float64 x
float64 y
float64 v
string change_left
string change_right
